flows.go

```diff
@@ -54,31 +54,18 @@ func (a Endpoint) LessThan(b Endpoint) bool {
 	return a.typ < b.typ || (a.typ == b.typ && bytes.Compare(a.raw[:a.len], b.raw[:b.len]) < 0)
 }
 
-// fnvHash is used by our FastHash functions, and implements the FNV hash
-// created by Glenn Fowler, Landon Curt Noll, and Phong Vo.
-// See http://isthe.com/chongo/tech/comp/fnv/.
-func fnvHash(s []byte) (h uint64) {
-	h = fnvBasis
-	for i := 0; i < len(s); i++ {
-		h ^= uint64(s[i])
-		h *= fnvPrime
-	}
-	return
-}
-
-const fnvBasis = 14695981039346656037
-const fnvPrime = 1099511628211
-
 // FastHash provides a quick hashing function for an endpoint, useful if you'd
 // like to split up endpoints by modulos or other load-balancing techniques.
-// It uses a variant of Fowler-Noll-Vo hashing.
+// It creates the hash by xor-ing the bytes of the address in a 64-bit, round robin
+// fashion (hash ^= addr[0],hash ^= addr[1]<<8,...,hash ^= addr[9]<<8). For
+// addresses less than or equal to 64-bits this guarantees  collision free hashes.
 //
 // The output of FastHash is not guaranteed to remain the same through future
 // code revisions, so should not be used to key values in persistent storage.
 func (a Endpoint) FastHash() (h uint64) {
-	h = fnvHash(a.raw[:a.len])
-	h ^= uint64(a.typ)
-	h *= fnvPrime
+	for i := 0; i < a.len; i++ {
```

Comment:
effectively same comments as below, here (sorry, I'm reviewing this upside down):

```
if a.len <= 8 {
  return binary.LittleEndian.Uint64(a.raw[:8])
}
// keep old fnv for larger values.
```

Reply:
see first thread